ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/parsers/StringUtils.kt

```diff
@@ -167,6 +173,33 @@ private fun String.validateSymbols(lineNo: Int) {
     }
 }
 
+private fun String.getCommentStartIndex(): Int {
+    // temporary replace \" and \' to hide non-significant quotes
+    // we cannot remove them because we need to know the original index
+    val chars = this
+        .replace("\\\"", "__")
+        .replace("\\\'", "__")
```

Comment:
The TOML spec allows backslashes to be interpreted literally in single-quoted strings, we just have a config option to enable single quote escaping. Would this work with, say, `a = 'C:\some\path\'`?

Reply:
Good point! It doesn't work for this test

Reply:
Fixed. You can see additional tests in StringUtilsTest, now they work